### PR TITLE
Feature/tank api

### PR DESCRIPTION
**Pull request by MsweliP**

### CODE REVIEW

File: WaterLog_Backend/Controllers/TankMonitorsController.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+using WaterLog_Backend.Models;
+
+
+namespace WaterLog_Backend.Controllers
+{
+        [Route("api/[controller]")]
+        [ApiController]
+        public class TankMonitorsController : ControllerBase
+        {
+        private readonly DatabaseContext _db;
```
Comment (ash-rabbit) [inline WaterLog_Backend/Controllers/TankMonitorsController.cs:20]: fix the indentation please, there should be a tab space in the class.

**Review by ash-rabbit — CHANGES_REQUESTED** [review]

File: WaterLog_Backend/Controllers/TankMonitorsController.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        readonly IConfiguration _config;
+        public TankMonitorsController(DatabaseContext context, IConfiguration config)
+        {
+            _db = context;
+            _config = config;
+        }
+        // GET api/segments
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<TankMonitorsEntry>>> Get()
```
Comment (ash-rabbit) [inline WaterLog_Backend/Controllers/TankMonitorsController.cs]: What are you getting, tank monitors? maybe rename your method to GetWhateverYouGetting

File: WaterLog_Backend/Controllers/TankMonitorsController.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        {
+            return await _db.TankMonitors.ToListAsync();
+        }
+
+        // GET api/TankMonitorById/
+        [HttpGet("{id}")]
+        public async Task<ActionResult<TankMonitorsEntry>> Get(int id)
```
Comment (ash-rabbit) [inline WaterLog_Backend/Controllers/TankMonitorsController.cs]: Rename the method GetWhateverYouGettingById

**Review by ash-rabbit — CHANGES_REQUESTED** [review]

File: WaterLog_Backend/Controllers/TankMonitorsController.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        public TankMonitorsController(DatabaseContext context, IConfiguration config)
+        {
+            _db = context;
+            _config = config;
+        }
+        // GET api/segments
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<TankMonitorsEntry>>> Get()
+        {
+            return await _db.TankMonitors.ToListAsync();
```
Comment (ash-rabbit) [inline WaterLog_Backend/Controllers/TankMonitorsController.cs]: What if the database connection fails? or the return is null?

File: WaterLog_Backend/Controllers/TankMonitorsController.cs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        }
+
+        // GET api/TankMonitorById/
+        [HttpGet("{id}")]
+        public async Task<ActionResult<TankMonitorsEntry>> Get(int id)
+        {
+            return await _db.TankMonitors.FindAsync(id);
```
Comment (ash-rabbit) [inline WaterLog_Backend/Controllers/TankMonitorsController.cs]: What if the database connection fails? or the return is null?

**Review by ash-rabbit — CHANGES_REQUESTED** [review]

File: WaterLog_Backend/Controllers/TankMonitorsController.cs (hunk 2 above)
Comment (MsweliP) [reply WaterLog_Backend/Controllers/TankMonitorsController.cs]: Done

File: WaterLog_Backend/Controllers/TankMonitorsController.cs (hunk 1 above)
Comment (MsweliP) [reply WaterLog_Backend/Controllers/TankMonitorsController.cs:20]: Done

File: WaterLog_Backend/Controllers/TankMonitorsController.cs (hunk 3 above)
Comment (MsweliP) [reply WaterLog_Backend/Controllers/TankMonitorsController.cs]: Done

File: WaterLog_Backend/Controllers/TankMonitorsController.cs (hunk 4 above)
Comment (MsweliP) [reply WaterLog_Backend/Controllers/TankMonitorsController.cs]: Done , please re-check

File: WaterLog_Backend/Controllers/TankMonitorsController.cs (hunk 5 above)
Comment (MsweliP) [reply WaterLog_Backend/Controllers/TankMonitorsController.cs]: Done

File: WaterLog_Backend/Controllers/TankMonitorsController.cs (hunk 2 above)
Comment (CilliersN) [reply WaterLog_Backend/Controllers/TankMonitorsController.cs]: 😂 bro.....

File: WaterLog_Backend/Controllers/TankMonitorsController.cs (hunk 5 above)
Comment (ash-rabbit) [reply WaterLog_Backend/Controllers/TankMonitorsController.cs]: remove the line with `return await _db.TankMonitors.FindAsync(id);`

**Review by ash-rabbit — CHANGES_REQUESTED** [review]

File: WaterLog_Backend/Controllers/TankMonitorsController.cs [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        // GET api/segments
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<TankMonitorsEntry>>> Get()
+        {
+            return await _db.TankMonitors.ToListAsync();
+        }
+
+        // GET api/TankMonitorById/
+        [HttpGet("{id}")]
+        public async Task<ActionResult<TankMonitorsEntry>> GetWhateverYouGetting(int id)
```
Comment (CilliersN) [inline WaterLog_Backend/Controllers/TankMonitorsController.cs]: 😂  bro...

public async Task<ActionResult < TankMonitorsEntry > > GetWhateverYouGetting(int id)
???

**Review by CilliersN — CHANGES_REQUESTED** [review]

File: WaterLog_Backend/Controllers/TankMonitorsController.cs (hunk 6 above)
Comment (MsweliP) [reply WaterLog_Backend/Controllers/TankMonitorsController.cs]: Lol fixed

**Review by FredJEhlers — APPROVED** [review]

**Review by DeodatusVermaak — APPROVED** [review]

File: WaterLog_Backend/Controllers/TankMonitorsController.cs (hunk 2 above)
Comment (ash-rabbit) [reply WaterLog_Backend/Controllers/TankMonitorsController.cs]: How is it done when its not renamed?

File: WaterLog_Backend/Controllers/TankMonitorsController.cs (hunk 4 above)
Comment (ash-rabbit) [reply WaterLog_Backend/Controllers/TankMonitorsController.cs]: Its not done

**Review by ash-rabbit — APPROVED** [review]

**Pull request merged**